Add howto section "Rotating Log Files" (#1732)
cocotb/cocotb: documentation/source/rotating_logger.rst

.. _rotating-logger:

> ***********************
> Rotating Disk Log Files
> ***********************

Reviewer: ```suggestion
*****************
Log File Rotation
*****************
```
I'm not sure "disk" is a useful word here.

Author: Done.

> The following is an example of how to support rotation of disk log files.

Reviewer: ```suggestion
The following is an example of how to support rotation of log files.
```

Author: Done.

> It will keep the newest 3 files which are at a maximum 5 MiB large.

Reviewer: ```suggestion
It will keep the newest 3 files which are at a maximum of 5 MiB in size.
```

Author: I need to lose the "at" in the sentence too, right?

Reviewer: Perhaps 'at most 5 MiB in size"

Author: Done.

> See also :ref:`logging-reference-section`.

Reviewer: ```suggestion
See also :ref:`logging-reference-section`, and the python docs for :class:`logging.RotatingFileHandler`.
```

Author: Done.

.. _rotating-logger:

******************
Rotating Log Files
******************

The following is an example of how to support rotation of log files.
It will keep the newest 3 files which are at most 5 MiB in size.

See also :ref:`logging-reference-section`,

> and the Python documentation for :class:`logging.RotatingFileHandler`.

Reviewer: ```suggestion
and the Python documentation for :class:`logging.handlers.RotatingFileHandler`.
```
